imap_processing/tests/codice/test_codice_l1a.py

```diff
@@ -228,17 +228,11 @@ def test_l1a_validate_data_arrays(test_l1a_data: xr.Dataset, index):
         validation_dataset = load_cdf(VALIDATION_DATA[index])
 
         for counter in counters:
-            # Ensure the data array shapes are equal
-            assert (
-                processed_dataset[counter].data.shape
-                == validation_dataset[counter].data.shape
+            # Ensure the data arrays are equal
+            np.testing.assert_equal(
+                processed_dataset[counter].data, validation_dataset[counter].data
```

Comment:
Give the xarray version a try, and I think you can drop the `.data` access as well.
```suggestion
            xr.testing.assert_equal(
                processed_dataset[counter], validation_dataset[counter]
```

Reply:
Cool, I wasn't aware of `xr.testing`!

Unfortunately we don't have quite the same metadata values (for example, Joey uses camelcase e.g. "EnergyTable" where i use lowercase + underscores e.g. "energy_table" (which I think is the SPDF preferred convention?)) so this will throw errors.

When I try to compare just the data arrays with `xr.testing`, I get an error `TypeError: <class 'numpy.ndarray'> not supported by assertion comparison`

I can get around this by converting the numpy arrays to `xr.DataArray` first:

```python
            xr.testing.assert_equal(
                xr.DataArray(processed_dataset[counter].data), xr.DataArray(validation_dataset[counter].data)
            )
```

Not sure its worth doing it this way though? It seems like its just doing the same thing as `np.testing` at this point.

Reply:
Agreed, `np.testing` is the way to go here then!